src/geo/transform.js

```diff
@@ -1524,7 +1524,7 @@ class Transform {
         //Calculate the offset of the tile
         const canonical = unwrappedTileID.canonical;
         const windowScaleFactor = 1 / this.height;
-        const scale = this.cameraWorldSize / this.zoomScale(canonical.z);
+        const scale = this.worldSize / this.zoomScale(canonical.z);
```

Comment:
Do we have any render tests for validating the fix? Definition for the `distance-from-center` expression is:
```
Returns the distance of a symbol instance from the center of the map.
The distance is measured in pixels divided by the height of the map container.
```
This change might suffer from same problems as the fog when you modified the other function to use worldSize instead of cameraWorldSize. I believe you should modify both `scale` and `center` variables to use pixel coordinates of the cameraWorldSize rather than worldSize.

Reply:
> Do we have any render tests for validating the fix? 

yes indeed it was in todo list above to add a render test.  

> This change might suffer from same problems as the fog when you modified the other function to use worldSize instead of cameraWorldSize.

In practice I think it's fine to use `worldSize` and leave the rest unchanged since we don't really care about terrain elevation (we use map center point) and so I also don't think this breaks the contract of the API. The rationale for me here was that we keep it as simple as possible and not involve terrain (cameraWorldSize varies with elevation) in the calculations in the first place. But I am curious why you think using `cameraWorldSize` and fixing `scale` and `center` instead is in practice a better solution?

Reply:
You're on a right track here but `worldSize` works quite the opposite: it can be interpret as a distance between the camera physical location and the map center point. With terrain enabled this distance can vary a lot especially in areas with big elevation differences because by default the camera keeps its altitude relative to sea level. Such changes in the default zoom level are also reflected in the `worldSize` which could be seen as the fog distance jumping back and forth. For this reason the connection between screen pixels and world pixels is lost making the result of the filter dependant on camera location and terrain elevation. 

The "camera zoom" was initially introduced as a zoom level that's computed relative to the sea level in order to solve this very problem. Purpose of the `cameraZoom` has been changed to describe distance to average visible elevation and a new variable `this._seaLevelZoom` has been introduced. I think we should consider using either camera zoom or sea level zoom as the fix.

I'm not saying that using the default world size is totally out of question, but there has to be clear understanding and reasoning behind that.

Reply:
Thanks @mpulkki-mapbox for the explanation. Indeed I wasn't aware of some of subtleties involved when terrain is enabled. I decided to only update the `center` local variable in `calculateDistanceTileData` to be in the correct coordinate space. This was the root cause. Alternatively wecould also use the `this._seaLevelZoom` as follows:

```
const ws = this._seaLevelZoom === null ? this.worldSize : this.tileSize * this.zoomScale(this._seaLevelZoom);
const scale = ws / this.zoomScale(canonical.z);
const unwrappedX = canonical.x + Math.pow(2, canonical.z) * unwrappedTileID.wrap;
const tX = unwrappedX * scale;
const tY = canonical.y * scale;

const center = this.point;
center.x *= ws / this.worldSize;
center.y *= ws / this.worldSize;
```

both of these are consistent.